src/librustc/diagnostics.rs

```diff
@@ -112,6 +112,50 @@ reference when using guards or refactor the entire expression, perhaps by
 putting the condition inside the body of the arm.
 "##,
 
+E0009: r##"
+In a pattern, all values that don't implement the `Copy` trait have to be bound
+the same way. The goal here is to avoid binding simultaneous by-move and by-ref
+at the same time.
+
+Wrong example:
+
+```
+struct X { x: (), }
+
+let x = Some((X { x: () }, X { x: () }));
+match x {
+    Some((y, ref z)) => {},
+    None => panic!()
+}
+```
+
+You have two solutions: either you implement the `Copy` trait on the X structure:
```

Comment:
This line is longer than 80 characters, might be best flowing across the next line:

```
There are two possible solutions.

1. Implement the `Copy` trait for the X structure:

2. Bind the pattern's values the same way:

(there's also a "the the" in the current version of the second case)